tests/components/airly/test_config_flow.py

```diff
@@ -0,0 +1,92 @@
+"""Define tests for the Airly config flow."""
+import json
+
+from asynctest import patch
+
+from homeassistant import data_entry_flow
+from homeassistant.const import CONF_API_KEY, CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.components.airly import config_flow
+from homeassistant.components.airly.const import DOMAIN
+
+from tests.common import load_fixture, MockConfigEntry
+
+CONFIG = {
+    CONF_NAME: "abcd",
+    CONF_API_KEY: "foo",
+    CONF_LATITUDE: 123,
+    CONF_LONGITUDE: 456,
+}
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    flow = config_flow.AirlyFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=None)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_invalid_api_key(hass):
+    """Test that errors are shown when API key is invalid."""
+    with patch(
+        "homeassistant.components.airly.config_flow.AirlyFlowHandler" "._test_api_key",
```

Comment:
We shouldn't patch our code but patch the library to raise the correct error:
https://codecov.io/gh/home-assistant/home-assistant/compare/0eb1d490467c40a6d46a6b94851548b1a5ef2eb8...cfaf576b312717ca7a4bbe290cdfc9bcf8a8d86e/diff#D13-97

Reply:
@MartinHjelmare I didn't know that. I have to patch aiohttp.ClientSession.get used in library and I have problems with that. Maybe there is some example in another components how to do that?

Reply:
We should be able to patch `"airly.MeasurementsSession.update".
```py
with patch("airly.MeasurementsSession.update", side_effect=AirlyError(...))
    ...
```